public/app/plugins/datasource/cloudwatch/ConfigEditor.tsx

```diff
@@ -0,0 +1,389 @@
+import React, { PureComponent, ChangeEvent } from 'react';
+import { DataSourcePluginOptionsEditorProps, FormLabel, Select, Input, Button } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import DatasourceSrv, { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import CloudWatchDatasource from './datasource';
+
+export type Props = DataSourcePluginOptionsEditorProps<any>;
+
+export interface State {
+  config: any;
+  authProviderOptions: SelectableValue[];
+  regions: SelectableValue[];
+}
+
+export class ConfigEditor extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { options } = this.props;
+
+    this.state = {
+      config: ConfigEditor.keyFill(options),
+      authProviderOptions: [
+        { label: 'Access & secret key', value: 'keys' },
+        { label: 'Credentials file', value: 'credentials' },
+        { label: 'ARN', value: 'arn' },
+      ],
+      regions: [],
+    };
+
+    this.datasourceSrv = getDatasourceSrv();
+
+    this.updateDatasource(this.state.config);
+
+    this.getRegions();
+  }
+
+  datasourceSrv: DatasourceSrv = null;
+
+  static getDerivedStateFromProps(props: Props, state: State) {
+    return {
+      ...state,
+      config: ConfigEditor.keyFill(props.options),
+    };
+  }
+
+  static keyFill = (options: any) => {
+    options.jsonData.authType = options.jsonData.authType || 'credentials';
+    options.jsonData.timeField = options.jsonData.timeField || '@timestamp';
+
+    if (!options.hasOwnProperty('editorSecureJsonData')) {
+      options.editorSecureJsonData = {
+        accessKey: '',
+        secretKey: '',
+      };
+    }
+
+    if (!options.hasOwnProperty('editorJsonData')) {
+      options.editorJsonData = {
+        assumeRoleArn: '',
+        customMetricsNamespaces: '',
+      };
+    }
+
+    if (!options.hasOwnProperty('secureJsonFields')) {
+      options.secureJsonFields = {
+        accessKey: false,
+        secretKey: false,
+      };
+    }
+
+    return options;
+  };
+
+  getRegions() {
```

Comment:
The old config editor had some sort of fallback for regions. I think in the case regions couldn't be loaded from the backend, a hard coded list of regions would be used. However I cannot see why that would be necessary since they were partially hard coded in the backend too. So I guess we can ignore that list. :)